### PR TITLE
Add a maintenance page

### PR DESCRIPTION
**Pull request by yjacolin**
* maintenance page template
* add an Apache config
* enable and disable in deploy hook
* doc in the deploy chapter

The maintenance page can be enable creating a maintenance.enable in apache
config dir after a build and an apache restart.

### CODE REVIEW

File: c2cgeoportal/scaffolds/create/apache/00_maintenance.conf.mako [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+RewriteCond %{REMOTE_ADDR} !^192\.168\.1\.1
+# Maintenance page
+Alias /maintenance ${directory}/${package}/static/maintenance/
+
+RewriteCond ${directory}/${package}/static/maintenance/maintenance.html -f
+RewriteCond ${directory}/apache/maintenance.enable -f
+
+RewriteCond %{SCRIPT_FILENAME} !maintenance/maintenance.html
+# Enable this line if you want to use image in html page
+# RewriteCond %{SCRIPT_FILENAME} !maintenance/maintenance.jpg
```
Comment (sbrunner) [inline c2cgeoportal/scaffolds/create/apache/00_maintenance.conf.mako]: jpeg ...

File: c2cgeoportal/scaffolds/create/apache/00_maintenance.conf.mako [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Document Root est htdocs
+RewriteEngine On
+# Localhost
+RewriteCond %{REMOTE_ADDR} !^127\.|::1
+RewriteCond %{REMOTE_HOST} !^localhost
+RewriteCond %{REMOTE_HOST} !^${host}
```
Comment (sbrunner) [inline c2cgeoportal/scaffolds/create/apache/00_maintenance.conf.mako]: host?

File: doc/integrator/deploy.rst [hunk 3]
```diff
@@ -227,3 +227,13 @@ Deploy on the prod server:
...
+
+
+
+Maintenance Page
+=================
+
+A maintenance page is enabled when starting the deploy process and is removed
+when the deploy ends.
+
+
```
Comment (sbrunner) [inline doc/integrator/deploy.rst]: Extra endlines ...

File: c2cgeoportal/scaffolds/create/apache/00_maintenance.conf.mako (hunk 2 above)
Comment (yjacolin) [inline c2cgeoportal/scaffolds/create/apache/00_maintenance.conf.mako]: I don't understand, we need to replace here the hostname for the url filter.

Comment (sbrunner) [inline c2cgeoportal/scaffolds/create/apache/00_maintenance.conf.mako]: Can you call me?

Comment (sbrunner) [inline c2cgeoportal/scaffolds/create/apache/00_maintenance.conf.mako]: OK for me :-)

**Comment by yjacolin** [comment]
Do not merge it yet, I want to improve the german translation.

**Comment by yjacolin** [comment]
@eleu we need your help for the german translation :)

File: c2cgeoportal/scaffolds/create/+package+/static/maintenance/maintenance.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+<article>
+    <h1>We&rsquo;ll be back soon!</h1>
+    <div>
+        <p>Sorry for the inconvenience but we&rsquo;re performing some maintenance at the moment.</p>
+    </div>
+</article>
+
+<article>
+    <h1>Seien Sie bald wieder!</h1>
```
Comment (eleu) [inline c2cgeoportal/scaffolds/create/+package+/static/maintenance/maintenance.html]: Wartungsarbeiten

File: c2cgeoportal/scaffolds/create/+package+/static/maintenance/maintenance.html [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    <div>
+        <p>Die Webseite wird derzeit Wartung. Vielen Dank für Ihre Geduld.</p>
```
Comment (eleu) [inline c2cgeoportal/scaffolds/create/+package+/static/maintenance/maintenance.html]: Die Webseite ist aufgrund von Wartungsarbeiten temporär nicht erreichbar. Bitte haben Sie Verständnis und versuchen Sie es später noch einmal.

**Comment by yjacolin** [comment]
Thanks @eleu

**Comment by yjacolin** [comment]
I don't know why travis failed here :( https://travis-ci.org/camptocamp/c2cgeoportal/builds/237548510#L4861

**Comment by ger-benjamin** [comment]
@yjacolin can you rebase and retry ?
Thanks

**Comment by yjacolin** [comment]
thanks @ger-benjamin !

**Comment by sbrunner** [comment]
Don't you want to merge it in 2.2?

**Comment by yjacolin** [comment]
@sbrunner do you agree that this PR is not more useful if we go toward the use of Docker?

**Comment by sbrunner** [comment]
Effectively ...

**Pull request closed without merging**